### ISSUE TITLE
deps.json for ProjectReferences is not emitted correctly

### ISSUE THREAD

**Issue by vatsan-madhavan** [dotnet/sdk#3254]
When I create a project with a project reference like this...

```XML
<ProjectReference Include="..\WindowsBase.csproj" />
```

... the `deps.json` file that's generated has an entry like this:

```json
      "WindowsBase/1.0.0": {
        "runtime": {
          "WindowsBase.dll": {}
        }
      }
```
The application loads `WindowsBase` from `Microsoft.NetCore.App` instead of loading the one supplied by the referenced project. 

[Repro: XUnitTestProject1.zip](https://github.com/dotnet/sdk/files/3205090/XUnitTestProject1.zip)

/cc @ericstj

**Comment by vatsan-madhavan** [comment]
Also related: #2674

**Comment by ericstj** [comment]
You can contrast this to a raw file reference which includes fileversion / assemblyversion.  For some reason fileversion and assemblyversion are missing from projectreferences.  This breaks the scenario where folks submodule a library to replace it (or in this case WPF just tries to test their own libs which in the case of WindowsBase overlap libs in the base shared framework).  /cc @dsplaisted

**Comment by weltkante** [comment]
Also happening in WinForms, deps files contain no version for facades referenced by test projects via ProjectReference and for `System.Drawing` that means it'll take the one supplied by `Microsoft.NetCore.App` instead of the one built in the WinForms repo. (Other facades for the various *.Design assemblies don't have a counterpart in Microsoft.NetCore.App as far as I can see.)

**Comment by ericstj** [comment]
I bet you can workaround by using both a project reference and a raw file reference.

**Comment by weltkante** [comment]
yeah having both kinds of references seems to be a useable workaround, loads the right System.Drawing now

**Comment by Tanya-Solyanik** [comment]
The following workaround https://github.com/dotnet/wpf/blob/main/eng/WpfArcadeSdk/tools/SdkReferences.targets#L141-#L167 when applies to the winforms [issue ](https://github.com/dotnet/winforms/issues/3106) about System.Drawing facade, solves the conflict resolution at build time between the project reference and the shared reference in favor of the local project reference, but the deps.json file still contains file version set to `0` instead of the 42.42... The resulting deps file does not work with runtime assembly loads for example getting this type fails because the assembly from Microsoft.NETCore.App is resolved and that one does not have the right type forward for this type. 

>var type = Type.GetType($"System.Drawing.Design.UITypeEditor, System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);

```
      "System.Drawing/10.0.0-dev": {     <<< correct version
        "dependencies": {
          "System.Configuration.ConfigurationManager": "10.0.0-preview.3.25155.12",
          "System.Windows.Extensions": "10.0.0-preview.3.25155.12",
          "System.Windows.Forms": "10.0.0-dev",
          "System.Windows.Forms.Design": "10.0.0-dev"
        },
        "runtime": {
          "System.Drawing.dll": {
            "fileVersion": "0.0.0.0"   <<<<<< should be 42...
          }
        }
```

Repro in ScratchProject - top commit in this branch  https://github.com/Tanya-Solyanik/winforms/tree/missing-facade-test
